packages/ckeditor5-font/src/ui/colortableview.ts

```diff
@@ -222,7 +162,59 @@ export default class ColorTableView extends View {
 			}
 		} );
 
-		this._documentColorsLabel = documentColorsLabel;
+		const colorGridComponentView = new ColorGridComponentView( locale, {
+			colors, columns, removeButtonLabel, documentColorsLabel, documentColorsCount, colorPickerLabel
+		} );
+
+		this.removeColorButtonView = colorGridComponentView.removeColorButtonView;
+
+		const colorPickerComponentView = new ColorPickerComponentView( locale );
+
+		this.colorTableComponent = colorGridComponentView;
+		this.colorPickerComponent = colorPickerComponentView;
+
+		this.set( 'isColorTableVisible', true );
+		this.set( 'isColorPickerVisible', false );
+
+		this.set( 'selectedColor', undefined );
+		this.set( 'originalColor', undefined );
+
+		const colorTableBind = Template.bind( this, this.colorTableComponent );
+		const colorPickerBind = Template.bind( this, this.colorPickerComponent );
+
+		this.colorTableComponent.extendTemplate( {
+			attributes: {
+				class: colorTableBind.if( 'isColorTableVisible', 'ck-hidden', value => !value )
+			}
+		} );
+
+		this.colorPickerComponent.extendTemplate( {
+			attributes: {
+				class: colorPickerBind.if( 'isColorPickerVisible', 'ck-hidden', value => !value )
+			}
+		} );
```

Comment:
This asks for `#isVisible` observable properties in `this.colorTableComponent` and `this.colorPickerComponent` bound internally to `ck-hidden` class. Then you would bind them here to `#isColorTableVisible` and `#isColorPickerVisible` respectively. 

This would create a chain of data (view model state) instead of cross-layer manipulation of templates. Let children components take care of themselves and encapsulate their internal logic. Communicate with them using a single property change.

https://en.wikipedia.org/wiki/Separation_of_concerns